src/tlo/methods/tb.py

```diff
@@ -2499,6 +2404,46 @@ def apply(self, person_id, squeeze_factor):
                     )
 
 
+class HSI_Tb_EndOfLifeCare(HSI_Event, IndividualScopeEventMixin):
+    """
+    this is a hospital stay for terminally-ill patients with TB
+    it does not affect disability weight or probability of death
+    no consumables are logged but health system capacity (HR) is allocated
+    there are no consequences if hospital bed is not available as person has scheduled death
+    already within 2 weeks
+    """
+
+    def __init__(self, module, person_id, beddays):
+        super().__init__(module, person_id=person_id)
+        assert isinstance(module, Tb)
+
+        self.TREATMENT_ID = "Tb_PalliativeCare"
+        self.EXPECTED_APPT_FOOTPRINT = self.make_appt_footprint({})
+        self.ACCEPTED_FACILITY_LEVEL = "2"
+
+        self.beddays = beddays
+        self.BEDDAYS_FOOTPRINT = (
+            self.make_beddays_footprint({"general_bed": self.beddays})
+            if self.beddays
+            else self.make_beddays_footprint({"general_bed": 7.5})
+        )
```

Comment:
I've added beddays=8 to the def__init__ statement

This can be over-written (I think) when HSI_Tb_EndOfLifeCare is called as the number of bed days is dependent on the scheduled date of death:

hsi_event=HSI_Tb_EndOfLifeCare(
                    person_id=person_id, module=self.sim.modules["Tb"], beddays=6
                )